can/io/logger.py

```diff
@@ -227,8 +228,8 @@ def _get_new_writer(self, filename: StringPathLike) -> FileIOMessageWriter:
             return cast(FileIOMessageWriter, logger)
         else:
             raise Exception(
-                "The Logger corresponding to the arguments is not a FileIOMessageWriter or "
-                "can.Printer"
+                f"The Logger, {logger.__class__.__name__}, corresponding to "
```

Comment:
The previous error message wasn't great. We should use simple language.

```python3
raise Exception(
    f"The log format \"{''.join(pathlib.Path(filename).suffixes[-2:])}\" "
    f"is not supported by {self.__class__.__name__}"
)
```
This results in `Exception: The log format ".db" is not supported by SizedRotatingLogger`.
Works for all subclasses, too. We could also point the user to the documentation but that might be a bit redundant.